sql/hive/src/test/scala/org/apache/spark/sql/hive/LogicalPlanToSQLSuite.scala

```diff
@@ -105,6 +105,27 @@ class LogicalPlanToSQLSuite extends SQLBuilderTest with SQLTestUtils {
     checkHiveQl("SELECT id FROM t0 UNION ALL SELECT CAST(id AS INT) AS id FROM t0")
   }
 
+  test("union distinct") {
+    checkHiveQl("SELECT * FROM t0 UNION SELECT * FROM t0")
+  }
+
+  // Parser is unable to parse the following query:
+  // SELECT  `u_1`.`id`
+  // FROM (((SELECT  `t0`.`id` FROM `default`.`t0`)
+  // UNION ALL (SELECT  `t0`.`id` FROM `default`.`t0`))
+  // UNION ALL (SELECT  `t0`.`id` FROM `default`.`t0`)) AS u_1
+  ignore("three-child union") {
```

Comment:
Thank you! @hvanhovell 

Based on the comment by @rxin , I think we still need a nesting version here for future extension, instead hacking the SQL generation. 

Saw @viirya did a similar work last month: https://github.com/apache/spark/pull/10660. Maybe @viirya can do me a favor for completing this? Thank you! : )

Also cc @liancheng